### ISSUE TITLE
pivot_table "row" keyword not supported by pandas 0.16.0

### ISSUE THREAD

**Issue by updiversity** [yhat/ggplot#417]
Using pandas 0.16.0 and ggplot 0.6.5, when using the following command:

```
from ggplot import *
p = ggplot(aes(x='total_amount'), data=alldata) + geom_histogram() + ggtitle('My Histogram')
p
```

I got the following error :

```
/share/venvs/analytics/local/lib/python2.7/site-packages/ggplot/stats/stat_bin.pyc in _calculate(self, data)
    125                             })
    126         _wfreq_table = pd.pivot_table(_df, values='weights',
--> 127                                       rows=['assignments'], aggfunc=np.sum)
    128 
    129         # For numerical x values, empty bins get have no value

TypeError: pivot_table() got an unexpected keyword argument 'rows'
```

According to pandas documentation, it looks like the "rows" keyword is not anymore supported : 
DataFrame.pivot_table and crosstab‘s rows and cols keyword arguments were removed in favor of index and columns (https://github.com/pydata/pandas/issues/6581)

**Comment by has2k1** [comment]
So far I fixed that in PR #360, however it is still a work in progress. You can fix it and submit a PR. Pandas 0.15.0 made a bunch of API changes and the current ggplot master branch has not been tested against it.

**Comment by stonebig** [comment]
Is there an official "abandonware" mark on this project ?

**Comment by FayolChang** [comment]
It seems that development of this package has been stopped.

**Comment by chris-barry** [comment]
I also get this error with every single example on http://ggplot.yhathq.com/docs/ 

Is there any workaround?

**Comment by Riptawr** [comment]
got it working by changing "rows" to "index" in ggplot/stats/stat_bin.py, on line 126-127:
`_wfreq_table = pd.pivot_table(_df, values='weights', rows=['assignments'], aggfunc=np.sum)`

**Comment by yyl** [comment]
After changing the code, how to make it work, build ggplot again?

**Comment by JanSchulz** [comment]
just restart the interpreter and it should work

**Comment by yyl** [comment]
Got it. thanks!

**Comment by dalanmiller** [comment]
Just got this error in my Jupyter:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/Users/dalanmiller/anaconda/lib/python3.4/site-packages/IPython/core/formatters.py in __call__(self, obj)
    695                 type_pprinters=self.type_printers,
    696                 deferred_pprinters=self.deferred_printers)
--> 697             printer.pretty(obj)
    698             printer.flush()
    699             return stream.getvalue()

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    381                             if callable(meth):
    382                                 return meth(obj, self, cycle)
--> 383             return _default_pprint(obj, self, cycle)
    384         finally:
    385             self.end_group()

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/IPython/lib/pretty.py in _default_pprint(obj, p, cycle)
    501     if _safe_getattr(klass, '__repr__', None) not in _baseclass_reprs:
    502         # A user-provided repr. Find newlines and replace them with p.break_()
--> 503         _repr_pprint(obj, p, cycle)
    504         return
    505     p.begin_group(1, '<')

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    683     """A pprint that just redirects to the normal repr function."""
    684     # Find newlines and replace them with p.break_()
--> 685     output = repr(obj)
    686     for idx,output_line in enumerate(output.splitlines()):
    687         if idx:

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/ggplot/ggplot.py in __repr__(self)
    109     def __repr__(self):
    110         """Print/show the plot"""
--> 111         figure = self.draw()
    112         # We're going to default to making the plot appear when __repr__ is
    113         # called.

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/ggplot/ggplot.py in draw(self)
    305 
    306                     data = self._make_plot_data(data, _aes)
--> 307                     callbacks = geom.plot_layer(data, ax)
    308                     if callbacks:
    309                         for callback in callbacks:

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/ggplot/geoms/geom.py in plot_layer(self, data, ax)
    115         _cols = set(data.columns) & set(self.manual_aes)
    116         data = data.drop(_cols, axis=1)
--> 117         data = self._calculate_stats(data)
    118         self._verify_aesthetics(data)
    119         _needed = self.valid_aes | self._extra_requires

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/ggplot/geoms/geom.py in _calculate_stats(self, data)
    276                 new_data = new_data.append(_data, ignore_index=True)
    277         else:
--> 278             new_data = self._stat._calculate(data)
    279 
    280         return new_data

/Users/dalanmiller/anaconda/lib/python3.4/site-packages/ggplot/stats/stat_bin.py in _calculate(self, data)
    125                             })
    126         _wfreq_table = pd.pivot_table(_df, values='weights',
--> 127                                       rows=['assignments'], aggfunc=np.sum)
    128 
    129         # For numerical x values, empty bins get have no value

TypeError: pivot_table() got an unexpected keyword argument 'rows'
```

**Comment by georgezoto** [comment]
Keep in mind that if you run into problems creating pivot tables (pivot_table "row" keyword not supported by pandas 0.16.0 #417) 
you should use the new syntax of index and columns instead of rows and cols. 

Also if you run into Pandas SettingWithCopyWarning then try to follow these steps:
myindex = mydataframe['mycolumn'] operation constant
mydataframe.loc[myindex, 'mycolumn'] = new value

See also:
http://stackoverflow.com/questions/26724378/pandas-settingwithcopywarning
http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

**Comment by glamp** [comment]
fixed in 0.9.3
